engine/flink/components/base-tests/src/test/scala/pl/touk/nussknacker/engine/flink/table/sink/SinkWriterWithExceptionHandlerTest.scala

```diff
@@ -0,0 +1,127 @@
+
+import com.typesafe.config.ConfigFactory
+import org.apache.flink.api.common.functions.FlatMapFunction
+import org.apache.flink.api.connector.sink2.{Sink, SinkWriter, WriterInitContext}
+import org.scalatest.{BeforeAndAfterAll, LoneElement}
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.engine.api.{Context, LazyParameter, MethodToInvoke, ParamName, ValueWithContext}
+import pl.touk.nussknacker.engine.api.component.{ComponentDefinition, NodeComponentInfo}
+import pl.touk.nussknacker.engine.api.process.SinkFactory
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.flink.api.exception.{ExceptionHandler, SinkWriterWithExceptionHandler}
+import pl.touk.nussknacker.engine.flink.api.process.{
+  BasicFlinkSinkV2,
+  FlinkCustomNodeContext,
+  FlinkLazyParameterFunctionHelper
+}
+import pl.touk.nussknacker.engine.flink.minicluster.FlinkMiniClusterFactory
+import pl.touk.nussknacker.engine.flink.table.utils.ModelClassLoaderSimulationSuite
+import pl.touk.nussknacker.engine.flink.util.test.FlinkTestScenarioRunner
+import pl.touk.nussknacker.engine.flink.util.test.FlinkTestScenarioRunner.FlinkTestScenarioRunnerExt
+import pl.touk.nussknacker.engine.process.FlinkJobConfig.ExecutionMode
+import pl.touk.nussknacker.engine.process.helpers.TestResultsHolder
+import pl.touk.nussknacker.engine.spel.SpelExtension.SpelExpresion
+import pl.touk.nussknacker.engine.util.test.TestScenarioRunner
+import pl.touk.nussknacker.test.{PatientScalaFutures, ValidatedValuesDetailedMessage}
+
+import scala.annotation.nowarn
+
+class SinkWriterWithExceptionHandlerTest
+    extends AnyFunSuite
+    with Matchers
+    with PatientScalaFutures
+    with LoneElement
+    with ValidatedValuesDetailedMessage
+    with BeforeAndAfterAll
+    with ModelClassLoaderSimulationSuite {
+
+  private lazy val sinkComponent: ComponentDefinition = ComponentDefinition(name = "sinkV2", component = CollectSink)
+  private lazy val flinkMiniClusterWithServices       = FlinkMiniClusterFactory.createUnitTestsMiniClusterWithServices()
+
+  private lazy val runner: FlinkTestScenarioRunner =
+    TestScenarioRunner
+      .flinkBased(ConfigFactory.empty(), flinkMiniClusterWithServices)
+      .withExecutionMode(ExecutionMode.Streaming)
+      .withExtraComponents(List(sinkComponent))
+      .build()
+
+  override protected def afterAll(): Unit = {
+    super.afterAll()
+    flinkMiniClusterWithServices.close()
+  }
+
+  test("new source handle exceptions properly when using SinkWriterWithExceptionHandler trait") {
+    val scenario = ScenarioBuilder
+      .streaming("test")
+      .source("source", "source")
+      .emptySink("sinkV2", "sinkV2", "value" -> "#input".spel)
+
+    val result = runner.runWithData(scenario, List(1, 2, 3, 0, 4))
+    result.validValue.errors shouldBe empty
```

Comment:
Why are errors empty? I expected that there would be one error with dividing by zero problem